crates/wasmtime/src/runtime/module.rs

```diff
@@ -343,14 +345,35 @@ impl Module {
     /// state of the file.
     #[cfg(all(feature = "std", any(feature = "cranelift", feature = "winch")))]
     pub unsafe fn from_trusted_file(engine: &Engine, file: impl AsRef<Path>) -> Result<Module> {
-        let mmap = MmapVec::from_file(file.as_ref())?;
+        let open_file = open_file_for_mmap(file.as_ref())?;
+        Self::from_trusted_open_file(engine, open_file, Some(file.as_ref()))
+    }
+
+    /// A version of [`from_trusted_file`] that takes an open [`File`] instead
+    /// of a [`Path`]. This can be useful in situations where the `wasmtime`
+    /// process doesn't have full permissions to access the file system.
+    ///
+    /// # Unsafety
+    ///
+    /// All the of the reasons [`from_trusted_file`] is `unsafe` apply here as
+    /// well.
+    ///
+    /// [`from_trusted_file`]: Module::from_trusted_file
+    ///
+    #[cfg(all(feature = "std", any(feature = "cranelift", feature = "winch")))]
+    pub unsafe fn from_trusted_open_file(
+        engine: &Engine,
+        file: File,
+        wasm_path: Option<&Path>,
```

Comment:
It is not clear to me if this argument needs to match the actual location of the file you pass in or not. Or what happens when it doesn't match.

Reply:
Yeah I'm still cleaning things up a bit. It's there so that we can still use it when calling from `from_trusted_file`, but I think we can remove it from the public version of this one.